tests/standalone/connectiontimeoutexception-001.phpt

```diff
@@ -5,6 +5,7 @@ ConnectionTimeoutException: exceeding sockettimeoutms
 <?php skip_if_not_live(); ?>
 <?php skip_if_not_clean(); ?>
 <?php skip_if_test_commands_disabled(); ?>
+<?php skip_if_sleep_command_unavailable(); ?>
```

Comment:
Since we already have a skip for disabled test commands on the line above, what is the case where [`sleep`](https://mongodb-documentation.readthedocs.io/en/latest/reference/command/sleep.html) isn't supported? Unfortunately, it looks like the command is no longer in our own manual, so I've linked to one of the Chinese translations (cloned from an older version).

Reply:
Interesting. I wonder if that's related to fail points on mongos not being supported. Depending on the outcome of that ticket, it may make sense to simply skip this test for mongos and leave `/* See: SERVER-36066 */` as a comment between the skip function and `?>`.

Reply:
The work for skipping this properly (based on the sleep function existing) is already done, so I don't see a reason to remove that again.

Reply:
If I understand `skip_if_sleep_command_unavailable()` correctly, it's going to introduce a one-second delay if the server _does_ support the command. If this is also the only place that function would be used, I think the skip is preferable.

I think the best improvement might actually be modifying `skip_if_test_commands_disabled()` to also skip if the primary server is a `mongos` node, since it doesn't seem to support any test commands (despite reporting otherwise when we query its configuration).

That will also prove more useful down the line if we inevitably add more tests that want to set fail points.

I'm actually curious how the following tests pass on mongos:

 *  [`tests/standalone/executiontimeoutexception-001.phpt`](https://github.com/mongodb/mongo-php-driver/blob/1.5.1/tests/standalone/executiontimeoutexception-001.phpt)
 *  [`tests/standalone/executiontimeoutexception-002.phpt`](https://github.com/mongodb/mongo-php-driver/blob/1.5.1/tests/standalone/executiontimeoutexception-002.phpt)

The other two tests that set fail points are currently disabled:

 * [`tests/cursor/cursor-getmore-005.phpt`](https://github.com/mongodb/mongo-php-driver/blob/1.5.1/tests/cursor/cursor-getmore-005.phpt)
 * [`tests/cursor/cursor-getmore-006.phpt`](https://github.com/mongodb/mongo-php-driver/blob/1.5.1/tests/cursor/cursor-getmore-006.phpt)

Reply:
I can't explain why those two tests actually work, but I checked, and they indeed fail with ``operation exceeded time limit``, even when talking to ``mongos``. I can't reproduce this effect in the shell though.

Reply:
And also, setting the failpoint on ``mongos`` succeeds:

```
mongos> db.runCommand( { configureFailPoint: 'maxTimeAlwaysTimeOut', mode: 'alwaysOn' } );
{
	"ok" : 1,
	"operationTime" : Timestamp(1531733041, 1),
	"$clusterTime" : {
		"clusterTime" : Timestamp(1531733041, 1),
		"signature" : {
			"hash" : BinData(0,"AAAAAAAAAAAAAAAAAAAAAAAAAAA="),
			"keyId" : NumberLong(0)
		}
	}
}
```

Reply:
Ah, so it is a case of this specific test command not working on mongos. Works for me, then.